tests/php/ORM/DataListEagerLoadingTest.php

```diff
@@ -185,186 +185,216 @@ public function testEagerLoadRelations(string $iden, array $eagerLoad, int $expe
         $this->assertSame($expected, $selectCount);
     }
 
+    public function testEagerLoadRelationsCached(): void
+    {
+        // This test only works with "all" because the others lazy-load some of the data
+        // which is done in separate uncached lists
+        $scenario = $this->provideEagerLoadRelations()['all'];
+        $eagerLoad = $scenario['eagerLoad'];
+        $expected = $scenario['expected'];
+        $this->createEagerLoadData();
+
+        // First time will be executed in the db
+        $dataList = EagerLoadObject::get()->setUseCache(true)->eagerLoad(...$eagerLoad);
+        list($results, $selectCount) = $this->iterateEagerLoadData($dataList);
+        $expectedResults = $this->expectedEagerLoadRelations();
+        // Sort because the order of the results doesn't really matter - and has proven to be different in postgres
+        sort($expectedResults);
+        sort($results);
+        $this->assertSame($expectedResults, $results);
+        $this->assertSame($expected, $selectCount);
+
+        // Second time will use cached results - should be 0 queries with the same results
+        $dataList = EagerLoadObject::get()->setUseCache(true)->eagerLoad(...$eagerLoad);
+        list($results, $selectCount) = $this->iterateEagerLoadData($dataList);
+        $expectedResults = $this->expectedEagerLoadRelations();
+        // Sort because the order of the results doesn't really matter - and has proven to be different in postgres
+        sort($expectedResults);
+        sort($results);
+        $this->assertSame($expectedResults, $results);
+        $this->assertSame(0, $selectCount);
+    }
+
     public static function provideEagerLoadRelations(): array
     {
         return [
             // Include the lazy-loaded expectation here, since if the number
             // of queries changes for this we should expect the number
             // to change for eager-loading as well.
-            [
+            'lazy-load' => [
```

Comment:
Needed to add keys since `testEagerLoadRelationsCached` needs to grab specifically only the `all` scenario.

Can't remove `iden` because that's actually used in some tests that use this data provider.